contracts/system/NerifToken.sol

```diff
@@ -0,0 +1,39 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.18;
+
+import "@openzeppelin/contracts-upgradeable/token/ERC20/extensions/ERC20PermitUpgradeable.sol";
+
+import "./ContractKeys.sol";
+import "../common/ContractRegistry.sol";
+import "../interfaces/IMintableToken.sol";
+
+contract NerifToken is IMintableToken, ERC20PermitUpgradeable, ContractKeys {
+    ContractRegistry public contractsRegistry;
+
+    modifier onlyTokensVesting() {
+        _onlyTokensVesting();
+        _;
+    }
+
+    function initialize(
+        address _contractsRegistry,
+        uint256 _initTokensAmount,
+        string memory _name,
+        string memory _symbol
+    ) external initializer {
+        __ERC20_init(_name, _symbol);
+        __ERC20Permit_init(_name);
+
+        contractsRegistry = ContractRegistry(_contractsRegistry);
+
+        _mint(msg.sender, _initTokensAmount);
+    }
+
+    function mint(address _recipient, uint256 _tokensAmount) external override onlyTokensVesting {
```

Comment:
Does this mean that ONLY the vesting contract allowed to mint tokens?

Reply:
Yes, in the current implementation only the TokensVesting contract has the ability to create new tokens